sql-cli/examples/_dags/advanced.py

```diff
@@ -0,0 +1,69 @@
+# NOTE: This is an auto-generated file. Please do not edit this file manually.
+from airflow import DAG
+from airflow.utils import timezone
+from airflow.utils.task_group import TaskGroup
+from astro import sql as aql
+from astro.sql.table import Metadata, Table
+
+with DAG(
+    dag_id="advanced",
+    start_date=timezone.parse("2022-09-20"),
+) as dag:
+    sub_dir_1 = TaskGroup(group_id="sub_dir_1")
+    sub_dir_2 = TaskGroup(group_id="sub_dir_2")
+    sub_dir_1_1 = TaskGroup(group_id="sub_dir_1_1")
+    with sub_dir_1, sub_dir_1_1:
+        sub_dir_1_sub_dir_1_1_b = aql.transform_file(
+            file_path="""
+                select 1
+            """,
+            parameters={
+            },
+            op_kwargs={
+                "output_table": Table(
+                    name="b",
+                    metadata=Metadata(
+                        schema="sub_dir_1_1",
+                    ),
+                ),
+            },
+            task_id="b",
+        )
+    with sub_dir_1:
+        sub_dir_1_a = aql.transform_file(
+            file_path="""
+                select * from {{ sub_dir_1_sub_dir_1_1_b }}
+            """,
+            parameters={
+                "sub_dir_1_sub_dir_1_1_b": sub_dir_1_sub_dir_1_1_b,
+            },
+            op_kwargs={
+                "output_table": Table(
+                    name="a",
+                    metadata=Metadata(
+                        schema="sub_dir_1",
+                    ),
+                ),
+            },
+            task_id="a",
+        )
+    with sub_dir_2:
+        sub_dir_2_a = aql.transform_file(
+            file_path="""
+                select * from {{ sub_dir_1_a }}
+                union
+                select * from {{ non_existing_reference }}
+            """,
+            parameters={
+                "sub_dir_1_a": sub_dir_1_a,
+            },
+            op_kwargs={
+                "output_table": Table(
+                    name="a",
```

Comment:
why are we outputing to a table named "a"? Either the tables should have names that mean something or we should use temporary tables.

Reply:
Because currently, the table name equals the file name as in dbt. I am completely open to use sql headers or yaml for other and more configuration options.

Reply:
fixed in c6902aa